### ISSUE TITLE
TIMs not being received by HMI, Distribution Type field not being correctly encoded

### ISSUE THREAD

**Issue by mvs5465** [usdot-jpo-ode/jpo-ode#233]
## Overview

WyDOT reported that Traveler Information Messages to be distributed via satellite are not being received/displayed by the OBU HMI. These expected flow of these messages looks like: `ODE --> SDW --> OBU --> HMI`. We know from querying that the messages are successfully being received by the SDW. 

It was noticed that the `distType` field in the `AdvisoryDetails` ASN header is not being encoded as expected, this difference may cause messages to either not be distributed by the SDW or considered valid by the OBU.

## Additional Considerations
 
The R8 and R9 versions of the specification for Traveler Situation Data messages defines this field with different values. 

R8:
```
distributionType
--indicates   how the ASDM is distributed; 1=rsu, 2=ip, 4=satellite, 8=reserved
```

R9:
```
distributionType
--indicates   how the ASDM is distributed; 1=rsu, 2=ip
```

The latest Southeast Michigan (SEMI) J2735 extension ASN.1 specification that governs communication schema with the SDC/SDW defines values for this field are as follows:
```
DistributionType ::= OCTET STRING (SIZE(1))
	--	none		(0),	"00000000", not intended for redistribution
	--	rsu		(1),	"00000001", intended for redistribution over DSRC
	--	ip		(2),	"00000010" intended for redistribution over IP
```

It appears as though "satellite" would be the most appropriate option for this field, but this is not present in the most recent (R9) specification. 

**Additionally, WyDOT has reported that messages created and encoded outside of the ODE specify _distType to 1_, and those messages _do appear_ on the HMI. This is inconsistent with the documentation.**

## Technical Details

Messages sent to the ASN1c encoder are in XML form, and return as an encoded hex dump. When decoding the resultant hex, the issue can be seen:

Input:
`<distType>2</distType>`

Expected Output:
`distType '02'H,`

Actual Output:
`distType '20'H,`

Lev mentioned in a reply to [an issue with ASN1c reported on Stack Overflow](https://stackoverflow.com/questions/12141250/decoding-asn-1-compiler-output-as-strings) that OCTET STRING variable types are printed in byte order.

For example, an input integer value of 2 looks like this in binary form:

```0000 0010```

However, when encoded into hex, the two nibbles are read in opposite order:

```0010 0000```

This seems to be why an input value of 0x02 becomes an output value of (in hex) 0x20.

**Comment by mvs5465** [comment]
The ODE side of this issue has been fixed in PR #234. 

However, the SDW returns this error:
```
ERROR: gov.dot.its.jpo.sdcsdw.websocketsfragment.service.xerjsonparser.XerJsonParserBadTypeException: Could not parse 02 at path /AdvisorySituationData/asdmDetails/distType as a base-2 Integer
```

We believe this is a parsing mistake on their end, expecting the field to be binary when it is actually printed as a hexadecimal. 

I will leave this issue report open for now until we figure out more details about this error.

**Comment by mvs5465** [comment]
We've received confirmation that this issue has been fixed on the SDW side. Messages are successfully being deposited with the correct `distType`.
